moveit_ros/trajectory_cache/src/trajectory_cache.cpp

```diff
@@ -0,0 +1,1340 @@
+  query.appendGTE("workspace_parameters.min_corner.y", plan_request.workspace_parameters.min_corner.y);
+  query.appendGTE("workspace_parameters.min_corner.z", plan_request.workspace_parameters.min_corner.z);
+  query.appendLTE("workspace_parameters.max_corner.x", plan_request.workspace_parameters.max_corner.x);
+  query.appendLTE("workspace_parameters.max_corner.y", plan_request.workspace_parameters.max_corner.y);
+  query.appendLTE("workspace_parameters.max_corner.z", plan_request.workspace_parameters.max_corner.z);
+
+  // Joint state
+  //   Only accounts for joint_state position. Ignores velocity and effort.
+  if (plan_request.start_state.is_diff)
+  {
+    // If plan request states that the start_state is_diff, then we need to get
+    // the current state from the move_group.
+
+    // NOTE: methyldragon -
+    //   I think if is_diff is on, the joint states will not be populated in all
+    //   of our motion plan requests? If this isn't the case we might need to
+    //   apply the joint states as offsets as well.
+    //
+    // TODO: Since MoveIt also potentially does another getCurrentState() call
+    //   when planning, there is a chance that the current state in the cache
+    //   differs from the state used in MoveIt's plan.
+    //
+    //   When upstreaming this class to MoveIt, this issue should go away once
+    //   the class is used within the move group's Plan call.
+    moveit::core::RobotStatePtr current_state = move_group.getCurrentState();
+    if (!current_state)
+    {
+      RCLCPP_WARN(node_->get_logger(), "Skipping start query append: Could not get robot state.");
+      // *query = original;  // Undo our changes. (Can't. Copy not supported.)
+      return false;
+    }
+
+    moveit_msgs::msg::RobotState current_state_msg;
+    robotStateToRobotStateMsg(*current_state, current_state_msg);
+
+    for (size_t i = 0; i < current_state_msg.joint_state.name.size(); i++)
+    {
+      query.append("start_state.joint_state.name_" + std::to_string(i), current_state_msg.joint_state.name.at(i));
+      query_append_range_inclusive_with_tolerance(query, "start_state.joint_state.position_" + std::to_string(i),
+                                                  current_state_msg.joint_state.position.at(i), match_tolerance);
+    }
+  }
+  else
+  {
+    for (size_t i = 0; i < plan_request.start_state.joint_state.name.size(); i++)
+    {
+      query.append("start_state.joint_state.name_" + std::to_string(i), plan_request.start_state.joint_state.name.at(i));
+      query_append_range_inclusive_with_tolerance(query, "start_state.joint_state.position_" + std::to_string(i),
+                                                  plan_request.start_state.joint_state.position.at(i), match_tolerance);
+    }
+  }
+  return true;
+}
+
+bool TrajectoryCache::extract_and_append_trajectory_goal_to_query(
+    Query& query, const moveit::planning_interface::MoveGroupInterface& /* move_group */,
```

Comment:
I would leave it in case it is needed in the future, and also to keep all `extract_and_append`-type methods with the same signature of always passing in the move_group.

Let me know what you think?

Reply:
EDIT: Turns out I needed it after all.

See: https://github.com/moveit/moveit2/pull/2838/commits/806e0c528a96e345d0a4455affaf4457cfb6dbc4 , which catches an edge case where the cache is used without `move_group.setWorkspace()` being called.